src/stdlib_stats_distribution_normal.fypp

```diff
@@ -0,0 +1,332 @@
+                        re = r + x
+                        if( hz <= 0 ) re = -re
+                        exit L1
+                    end if L2
+                    x = hz * wn(iz)
+                    if( fn(iz) + uni(1.0_${k1}$) * (fn(iz-1) - fn(iz)) <        &
+                        exp(-HALF * x * x) ) then
+                        re = x
+                        exit L1
+                    end if
+
+                    hz = dist_rand(1_int32)
+                    iz = iand( hz, 127 )
+                    if( abs( hz ) < kn(iz) ) then
+                        re = hz * wn(iz)
+                        exit L1
+                    end if
+                end do L1
+            end if
+            res(i) = re * scale + loc
+        end do
+    end function rvs_norm_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    function rvs_norm_array_${t1[0]}$${k1}$(loc, scale, array_size) result(res)
+        ${t1}$, intent(in) :: loc, scale
+        integer, intent(in) :: array_size
+        integer :: i
+        ${t1}$ :: res(array_size)
+        real(${k1}$) :: tr, ti
+
+        do i = 1, array_size
+            tr = rvs_norm_r${k1}$(loc % re, scale % re)
+            ti = rvs_norm_r${k1}$(loc % im, scale % im)
+            res(i) = cmplx(tr, ti, kind=${k1}$)
+        end do
+    end function rvs_norm_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function pdf_norm_${t1[0]}$${k1}$(x, loc, scale) result(res)
+    !
+    ! Normal distribution probability density function
+    !
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+        ${t1}$, parameter :: sqrt_2_pi = sqrt(2.0_${k1}$ * acos(-1.0_${k1}$))
+
+        if(scale == 0._${k1}$) call error_stop("Error(pdf_norm): Normal"       &
+            //"distribution scale parameter must be non-zero")
+        res = exp(- 0.5_${k1}$ * (x - loc) * (x - loc) / (scale * scale)) /    &
+              (sqrt_2_Pi * scale)
+    end function pdf_norm_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    impure elemental function pdf_norm_${t1[0]}$${k1}$(x, loc, scale) result(res)
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+
+        res = pdf_norm_r${k1}$(x % re, loc % re, scale % re)
+        res = res * pdf_norm_r${k1}$(x % im, loc % im, scale % im)
+    end function pdf_norm_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function cdf_norm_${t1[0]}$${k1}$(x, loc, scale) result(res)
+    !
+    ! Normal distribution cumulative distribution function
+    !
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+        ${t1}$, parameter :: sqrt_2 = sqrt(2.0_${k1}$)
+
+        if(scale == 0._${k1}$) call error_stop("Error(cdf_norm): Normal"       &
+            //"distribution scale parameter must be non-zero")
+        res = (1.0_${k1}$ + erf((x - loc) / (scale * sqrt_2))) / 2.0_${k1}$
```

Comment:
While correct, this representation is prone to round-off error when the `erf` term is close to -1.0 - which in practice leads to greater relative errors when `x` is quite a few standard deviations below the mean.

To improve the results in that range of x, consider instead using the intrinsic `erfc`. I think the equation is `0.5*erfc(-(x-loc)/(scale*sqrt_2))`.

Reply:
Just now I tried changing this to be 
```
        !res = (1.0_${k1}$ + erf((x - loc) / (scale * sqrt_2))) / 2.0_${k1}$
        res = erfc(-(x - loc) / (scale * sqrt_2)) / 2.0_${k1}$
```
All the tests passed, and a comparison with R showed much better agreement when `x` was around -8 standard deviations below `loc`, where we expect the previous formulation to have trouble.

Reply:
Thanks.